### PR TITLE
Initial commit to kick off project

### PR DESCRIPTION
**Pull request by kungfoome**
- Creates a config using the init subcommand and tests login

### CODE REVIEW

File: pkg/netrc/reader.go
Comment (truppert-upstart) [inline pkg/netrc/reader.go:1]: Does this copyright allow copying code into your repo?  Also, why copy it instead of just using it as a module?  Did you have to make some tweaks to it?

File: api/client.go
Comment (truppert-upstart) [inline api/client.go:1]: the `/api` folder is usually used for OPENAPI (Swagger) docs.  This looks like it might belong nested under `/internal`

File: internal/cmd/root/help.go
Comment (truppert-upstart) [inline internal/cmd/root/help.go:1]: Was this meant to be in its own `help` package?

File: internal/query/query.go
Comment (truppert-upstart) [inline internal/query/query.go:11]: Golang is very opinionated on format.  You might want to run gofmt on your files.  I believe it would catch files without newlines like this one and fix them for you.

**Review by truppert-upstart — CHANGES_REQUESTED** [review]
Looks pretty good.  I had a few small comments in various files.

As a first pass, this seems perfectly fine.  I suspect you're going to want to refactor your utils package.  It's a bit of a catch-all right now.

Also, you may want to go back and add tests for your internal files.  There is quite a bit of code in there that could benefit from tests.

None of this is enough to prevent merging.  I'd just like quick answers to the few questions I asked before I approve if you don't mind.

File: pkg/netrc/reader.go
Comment (kungfoome) [reply pkg/netrc/reader.go:1]: > Looks pretty good. I had a few small comments in various files.
> 
> As a first pass, this seems perfectly fine. I suspect you're going to want to refactor your utils package. It's a bit of a catch-all right now.
> 
> Also, you may want to go back and add tests for your internal files. There is quite a bit of code in there that could benefit from tests.
> 
> None of this is enough to prevent merging. I'd just like quick answers to the few questions I asked before I approve if you don't mind.

Yeah, the tests are coming. I wanted to try a few things first. I'll probably still end refactoring some of the way you enter the code, but for now it's ok.

Comment (kungfoome) [reply pkg/netrc/reader.go:1]: > Does this copyright allow copying code into your repo? Also, why copy it instead of just using it as a module? Did you have to make some tweaks to it?

I dont think there is a module you can use like that. It's built into the go binary. See here https://go.dev/src/cmd/go/internal/auth/netrc.go

File: api/client.go
Comment (kungfoome) [reply api/client.go:1]: Not really. This is a normal pattern to interact with an API. internal should only be stuff thats related to the binary itself.  It's basically following this pattern https://github.com/cli/cli

File: internal/cmd/root/help.go
Comment (kungfoome) [reply internal/cmd/root/help.go:1]: root in this case is correct. It's also a common pattern since this is basically the default command to run when it doesn't recognize anything else.

File: internal/query/query.go
Comment (kungfoome) [reply internal/query/query.go:11]: Yeah, not sure how that got left out. It should be running on every save. Eventually I'm going to setup a tool called pre-commit to help prevent this.

File: api/client.go
Comment (truppert-upstart) [reply api/client.go:1]: I was going off of this standard: https://github.com/golang-standards/project-layout/tree/master/api

But it's not a point worth worrying about.

File: pkg/netrc/reader.go
Comment (truppert-upstart) [reply pkg/netrc/reader.go:1]: It might be good to cover your bases and just copy in the relevant lines from the governing [LICENSE](https://github.com/golang/go/blob/master/LICENSE) file for the original code.

File: api/client.go
Comment (kungfoome) [reply api/client.go:1]: Oh nice. I didn't even know that existed. Good call out though. I think it still fits that description though. This is how i've seen api used in most cases.

File: pkg/netrc/reader.go
Comment (kungfoome) [reply pkg/netrc/reader.go]: good idea. I added a license file and readme within that folder.

**Pull request merged**
